onadata/apps/api/management/commands/delete_users.py

```diff
@@ -0,0 +1,116 @@
+"""
+Delete users management command.
+"""
+import sys
+from django.contrib.auth.models import User
+from onadata.apps.logger.models import XForm
+from onadata.apps.logger.models import Instance
+from onadata.apps.logger.models import Project
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+
+def get_user_object_stats(username):  # pylint: disable=R0201
+    """
+    Get User information.
+    """
+    user = User.objects.get(username=username)
+    # Get the number of projects for this user
+    user_projects = Project.objects.filter(created_by=user).count()
+    # Get the number of forms
+    user_forms = XForm.objects.filter(user=user).count()
+    # Get the number of submissions
+    user_sumbissions = Instance.objects.filter(user=user).count()
+
+    result = {
+        'projects': user_projects,
+        'forms': user_forms,
+        'submissions': user_sumbissions
+    }
+
+    return result
+
+
+def inactivate_user(username):
+    """
+    Soft deletes the user termporarily.
+    """
+    try:
+        user = User.objects.get(username=username)
+        # set inactive status on user account
+        user.is_active = False
+        # append a timestamped suffix to the username
+        # to make the initial username available
+        soft_deletion_time = timezone.now()
+        deletion_suffix = soft_deletion_time.strftime('-deleted-at-%s')
+        user.username += deletion_suffix
+
+        user.save()
+        sys.stdout.write('User {} deleted successfully.'.format(username))
+        # confirm too that no user exists with provided email
+        try:
+            User.objects.get(email=user.email).delete()
+        except User.DoesNotExist:
+            pass
+    except User.DoesNotExist:
+        sys.stdout.write('User {} does not exist.'.format(username))
+
+
+class Command(BaseCommand):
+    """
+    Delete users management command.
+    """
+    help = 'Delete users'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--user', nargs='*')
+
+        parser.add_argument(
+            '--user_input',
+            action='store_true',
+            help='Confirm deletion of user account',
+            default=False
+        )
+
+    def handle(self, *args, **kwargs):
+        users = kwargs.get('user')
+        user_input = kwargs.get('user_input')
+
+        if users:
+            for user in users:
+                user_details = user.split(':')
+                username = user_details[0]
+                email = user_details[1]
+                if user_input is False:
+                    # If the --user_input flag is not provided.
+                    user_stats = get_user_object_stats(username)
```

Comment:
Suggestion: It would be nice if we printed the user stats no matter what.... Just in case someone doesn't want to make a choice in Line `89` but still wants to receive the overall summary of deleted objects

Reply:
Yeah, I've moved this.